packages/data/src/world/mq/ganon_castle_mq.yml

```diff
@@ -35,92 +39,254 @@
+    "MQ Ganon Castle Spirit Trial Back Right Sun Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+    "MQ Ganon Castle Spirit Trial Back Left Sun Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+    "MQ Ganon Castle Spirit Trial Front Left Sun Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+    "MQ Ganon Castle Spirit Trial Gold Gauntlets Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+"Ganon Castle Spirit Trial Ending Room":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Spirit Trial Sun Room Sun Side": "true"
+  events:
+    GANON_TRIAL_SPIRIT: "can_use_light_arrows"
+  locations:
+    "MQ Ganon Pot Spirit End 1": "true"
+    "MQ Ganon Pot Spirit End 2": "true"
+"Ganon Castle Shadow Trial Entrance Side":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Middle Beamos Area 1": "event(GANON_MQ_SHADOW_PROGRESSION_FROM_ENTRANCE)"
+    "Ganon Castle Main Room": "true"
+    "Ganon Castle Shadow Trial Ending Side": "climb_anywhere"
+  events:
+    GANON_MQ_SHADOW_BOMBFLOWER_FROM_ENTRANCE: "can_use_bow || ((longshot_anywhere || (has_lens && (has_hover_boots || hookshot_anywhere))) && (can_use_din || has_bombflowers))"
+    GANON_MQ_SHADOW_PROGRESSION_FROM_ENTRANCE: "longshot_anywhere || (has_lens && hookshot_anywhere) || (has_lens && ((can_hookshot && event(GANON_MQ_SHADOW_BOMBFLOWER_FROM_ENTRANCE)) || has_hover_boots))"
+  locations:
+    "MQ Ganon Castle Shadow Trial Bomb Flower Chest": "event(GANON_MQ_SHADOW_BOMBFLOWER_FROM_ENTRANCE) && (can_hookshot || has_hover_boots)"
+    "MQ Ganon Castle SR Shadow Front-Right": "event(GANON_MQ_SHADOW_PROGRESSION_FROM_ENTRANCE)"
+    "MQ Ganon Castle SR Shadow Front-Center": "event(GANON_MQ_SHADOW_PROGRESSION_FROM_ENTRANCE)"
+    "MQ Ganon Castle Wonder Item": "event(GANON_MQ_SHADOW_BOMBFLOWER_FROM_ENTRANCE) && event(GANON_MQ_SHADOW_PROGRESSION_FROM_ENTRANCE)"
+"Ganon Castle Shadow Trial Middle Beamos Area 1":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Entrance Side": "event(GANON_MQ_SHADOW_PROGRESSION_FROM_BEAMOS_1)"
+    "Ganon Castle Shadow Trial Middle Beamos Area 2": "event(GANON_MQ_SHADOW_TORCH_ICE_BLOCK) || has_hover_boots || hookshot_anywhere || (event(GANON_MQ_SHADOW_EYE_CHEST) && can_hookshot)"
+  events:
+    GANON_MQ_SHADOW_TORCH_ICE_BLOCK: "has_fire"
+    GANON_MQ_SHADOW_PROGRESSION_FROM_BEAMOS_1: "(has_lens && (can_longshot || hookshot_anywhere || (has_hover_boots && (is_adult || climb_anywhere)))) || longshot_anywhere"
+    GANON_MQ_SHADOW_BOMBFLOWER_FROM_BEAMOS_1: "can_use_bow || ((can_use_din || has_bombflowers) && (has_lens || longshot_anywhere))"
+  locations:
+    "MQ Ganon Castle SR Shadow Middle": "true"
+    "MQ Ganon Castle Shadow Trial Bomb Flower Chest": "event(GANON_MQ_SHADOW_BOMBFLOWER_FROM_BEAMOS_1) && (is_adult || has_hover_boots || hookshot_anywhere)"
+    "MQ Ganon Castle Wonder Item": "event(GANON_MQ_SHADOW_BOMBFLOWER_FROM_BEAMOS_1)"
```

Comment:
Since you'll actually need to get to the platform, this should require lens or longshot anywhere, since just shooting it with bow will not give you the item.

Reply:
Fixed.